syntax = "proto3";

service Calendar {
  rpc GetEvents(User) returns (stream Event) {}
  rpc CreateEvent(NewEvent) returns (Event) {}
}

message User {
  string id = 1;
}

message Event {
  string id = 1;
  float timestamp = 2;
  repeated string participant_ids = 3;
}

message NewEvent {
  repeated string participant_ids = 1;
}